File: Pancake_sorting.cpp
class Solution {
public:
    void flip(vector<int>& arr ,int  index){
        for(int i=0;i<=index/2;i++){
            int temp= arr[i];
            arr[i]=arr[index-i];
            arr[index-i]=temp;
        }
    }
    
    vector<int> pancakeSort(vector<int>& arr) {
        vector<int> ans;

        for(int i=arr.size()-1;i>0;i--){
            for(int j=1;j<=i;j++){
                if(arr[j]==i+1){
                    flip(arr,j);
                    ans.push_back(j+1);
                    break;
                }
            }
            flip(arr,i);
            ans.push_back(i+1);
        }

        return ans;
    }
};

File: pancake-sorting.py
class Solution:
    def pancakeSort(self, arr: List[int]) -> List[int]:
        res = []

        def flip(index):
            for i in range(0, index//2 + 1):
                temp = arr[i]
                arr[i] = arr[index-i]
                arr[index-i] = temp

        for i in range(len(arr)-1,0,-1):
            for j in range(1, i+1):
                if arr[j] == i+1:
                    flip(j)
                    res.append(j+1)
                    break
            flip(i)
            res.append(i+1)

        return res
        